### PR TITLE
Feature/docs

### PR DESCRIPTION
**Pull request by SRachamim**
Add descriptions and examples to each of the following functions in `Either`:

- `of`
- `foldMap`
- `reduce`
- `reduceRight`
- `flatten`
- `sequence`
- `traverse`

### CODE REVIEW

**Comment by raveclassic** [comment]
@gcanti This is great, let's merge

**Comment by gcanti** [comment]
Thanks @SRachamim

File: docs/modules/Either.ts.md [hunk 1]
```diff
@@ -250,32 +260,80 @@ Added in v2.0.0
...
 export declare const reduce: <A, B>(b: B, f: (b: B, a: A) => B) => <E>(fa: Either<E, A>) => B
 ```
 
+**Example**
+
+```ts
+import { pipe } from 'fp-ts/function'
+import * as E from 'fp-ts/Either'
+
+const empty = 'prefix'
```
Comment (gkamperis) [inline docs/modules/Either.ts.md:303]: as a noob I find this confusing vs the other "empty" like monoidString.empty above...
maybe a different var name?

File: docs/modules/Either.ts.md [hunk 2]
```diff
@@ -346,12 +404,24 @@ Added in v2.6.0
...
 ```ts
 export declare const flatten: <E, A>(mma: Either<E, Either<E, A>>) => Either<E, A>
 ```
 
+**Example**
+
+```ts
+import * as E from 'fp-ts/Either'
+
+assert.deepStrictEqual(E.flatten(E.right(E.right('a'))), E.right('a'))
```
Comment (gkamperis) [inline docs/modules/Either.ts.md:420]: as it is removing one level only do you think this is a good example?

assert.deepStrictEqual(E.flatten(E.right(E.right(E.right('a')))), E.right(E.right('a')))

File: docs/modules/Either.ts.md [hunk 3]
```diff
@@ -370,22 +440,51 @@ Added in v2.6.3
...
+**Example**
+
+```ts
+import { pipe } from 'fp-ts/function'
+import * as E from 'fp-ts/Either'
+import * as O from 'fp-ts/Option'
+
+assert.deepStrictEqual(pipe(E.right(O.some('a')), E.sequence(O.option)), O.some(E.right('a')))
+
+assert.deepStrictEqual(pipe(E.right(O.none), E.sequence(O.option)), O.none)
```
Comment (gkamperis) [inline docs/modules/Either.ts.md:460]: this baffles me as noob... if I can have a right of none then why can I not have a some of right of none?

also would be great if there was a left example....

**Pull request merged**